src/anyvar/restapi/main.py

```diff
@@ -145,6 +165,141 @@ def get_location_by_id(
+        AddAnnotationRequest,
+        Body(
+            description="Annotation to associate with the variation",
+        ),
+    ],
+) -> dict | HTTPException:
+    """Store an annotation for a variation.
+
+    :param request: FastAPI request object
+    :param vrs_id: the VRS ID of the variation to annotate
+    :param annotation: the annotation to store
+    :return: the variation and annotations if stored
+    :raise HTTPException: if requested location isn't found
+    """
+    messages = []
+    # Look up the variation from the AnyVar store
+    av: AnyVar = request.app.state.anyvar
+    try:
+        variation = av.get_object(vrs_id)
+    except KeyError as e:
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND, detail=f"Variation {vrs_id} not found"
+        ) from e
+
+    # Add the annotation to the annotation store
+    if hasattr(request.app.state, "anyannotation"):
+        anyannotation: AnyAnnotation = request.app.state.anyannotation
+        try:
+            anyannotation.put_annotation(
+                object_id=vrs_id,
+                annotation_type=annotation.annotation_type,
+                annotation=annotation.annotation,
+            )
+        except ValueError as e:
+            _logger.error("Failed to add annotation: %s", e)
+            raise HTTPException(
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                detail=f"Failed to add annotation: {annotation}",
+            ) from e
+
+    return {
+        "messages": messages,
+        "object": variation,
+        "object_id": vrs_id,
+        "annotation_type": annotation.annotation_type,
+        "annotation": annotation.annotation,
+    }
+
+
+@app.get(
+    "/variation/{vrs_id}/annotations/{annotation_type}",
+    response_model=GetAnnotationResponse,
+    response_model_exclude_none=True,
+    summary="Retrieve annotations for a variation",
+    description="Retrieve annotations for a variation by VRS ID and annotation type",
+    tags=[EndpointTag.VARIATIONS],
+)
+def get_variation_annotation(
+    request: Request,
+    vrs_id: Annotated[StrictStr, Path(..., description="VRS ID for variation")],
+    annotation_type: Annotated[StrictStr, Path(..., description="Annotation type")],
+) -> list[GetAnnotationResponse]:
+    """Retrieve annotations for a variation.
+
+    :param request: FastAPI request object
+    :param vrs_id: VRS ID for variation
+    :param annotation_type: type of annotation to retrieve
+    :return: list of annotations for the variation
+    """
+    # Retrieve the annotation from the annotation store
+    if hasattr(request.app.state, "anyannotation"):
+        anyannotation: AnyAnnotation = request.app.state.anyannotation
+        annotations = anyannotation.get_annotation(vrs_id, annotation_type)
+        # print("GOT ANNOTATIONS: \n" + str(annotations))
+    else:
+        annotations = []
+
+    return {"annotations": annotations}
+
+
+# TODO add input string annotation
```

Comment:
Can we remove this and either address in this PR or create a new issue to address later?

Reply:
https://github.com/biocommons/anyvar/issues/155